pkg/hintrunner/hinter/hinter.go

```diff
@@ -24,12 +24,13 @@ type HintRunnerContext struct {
 }
 
 func InitializeDefaultContext() *HintRunnerContext {
-	return &HintRunnerContext{
+	ctx := &HintRunnerContext{
 		DictionaryManager:         DictionaryManager{},
 		SquashedDictionaryManager: SquashedDictionaryManager{},
-		ScopeManager:              *DefaultNewScopeManager(),
 		ConstantSizeSegment:       mem.UnknownAddress,
 	}
+	InitializeScopeManager(ctx, make(map[string]any)) // Initialize the ScopeManager with a default scope
```

Comment:
well, now i look at it again, it seems that the current main branch returns `&HintRunnerContext` with `ScopeManager: *DefaultNewScopeManager(),` . This is actually the same as doing  `InitializeScopeManager(ctx, make(map[string]any))` after removing the ScopeManager field. 

@cicr99  what do you think? I'm not sure this is what was asked with issue #345  . This file is the hinter.go and the hinter effectively creates a new context with a main scope uninitialized.

Reply:
I guess the code that should be modified is in the `runHinterTests` function in zerohint_test.go  file? 

here : 

```
		ctx := &hinter.HintRunnerContext{}
		if tc.ctxInit != nil {
			tc.ctxInit(ctx)
		}
```
when running each test, the context is reset, with all its fields unitialized